app/src/main/java/com/teamagam/gimelgimel/app/mainActivity/view/MainActivityPanel.java

```diff
@@ -103,27 +109,57 @@ public void onPanelStateChanged(View panel, SlidingUpPanelLayout.PanelState prev
             sLogger.userInteraction("MainActivity's bottom panel mode changed from "
                     + previousState + " to " + newState);
             mViewModel.onChangePanelState(newState);
-        }
 
+            if (newState == SlidingUpPanelLayout.PanelState.ANCHORED) {
+                onPanelAnchored();
+            } else if (newState == SlidingUpPanelLayout.PanelState.COLLAPSED) {
+                onPanelCollapsed();
+            }
+        }
 
-        private void updateBottomPanelDimensions(float slideOffset) {
-            int height = calculateHeight(slideOffset);
-            adjustHeight(height);
+        private void updateBottomPanelDimensions(float bottomPanelSlideOffset) {
+            adjustBottomPanelContentHeight(getScreenWithoutCollapsedPanelHeight(),
+                    bottomPanelSlideOffset);
         }
 
-        private int calculateHeight(final float slideOffset) {
+        private int getScreenWithoutCollapsedPanelHeight() {
             int layoutHeight = mSlidingLayout.getHeight();
             int panelHeight = mSlidingLayout.getPanelHeight();
-            int calculatedHeight = (int) ((layoutHeight - panelHeight) * slideOffset);
-            int minimumHeight = (int) ((layoutHeight - panelHeight) * mSlidingLayout.getAnchorPoint());
+            return layoutHeight - panelHeight;
+        }
+
+        private void adjustBottomPanelContentHeight(int screenWithoutCollapsedPanelHeight,
+                                                    float slideOffset) {
+            int newHeight = (int) (screenWithoutCollapsedPanelHeight * slideOffset);
```

Comment:
It doesn't make sense to me. So screenWithoutCollapsedPanelHeight is indeed a constant? I mean, it's not changing with the sliding panel to reflect its *current* height? And if it is- so what is the math behind the equation:
newH = screenHwithoutPanel * offset ?